Add picture addition test

ivy-zhang opened the pull request:
1.add picture_addition_kernel{m}x{n}_{sse,sse2}_intrin test
2.add picture_addition_kernel{m}x{n}_av1_sse2_intrin test
3.add picture_addition_kernel16bit_sse2_intrin test

ivy-zhang committed:
Add unit tests for picture addition functions

1.add unit tests for picture_addition_kernel{m}x{n}_{sse,sse2}_intrin,
  picture_addition_kernel{m}x{n}_av1_sse2_intrin,
  picture_addition_kernel16bit_sse2_intrin test

ivy-zhang commented:
dry_run_test @./Bin/Release/SvtAv1UnitTests --gtest_filter=*PictureAddition*

ivy-zhang commented:
# Add Picture addition test

## Comments
## warnings in Local build
1. linux:  ok
2. windows:  ok

## warnings on remote ci servers
1. jenkins:
a. clang-format
b. 
2. travis
  a.  linux: ok
3. appvedor
  a. no warnings

## Test Plan: 
  Run the unit test with `./SvtAv1UnitTest --gtest_filter="*PictureAddition*"`, and check the result

## dry run output
~~~
$ ./SvtAv1UnitTests.exe --gtest_filter="*PictureAddition*"
Running main() from C:\Users\Ryan\addPic\SVT-AV1\third_party\googletest\src\gtest_main.cc
Note: Google Test filter = *PictureAddition*
[==========] Running 30 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 30 tests from PictureAddition/PictureAdditionTest
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAdditionTest/0
[       OK ] PictureAddition/PictureAdditionTest.PictureAdditionTest/0 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAdditionTest/1
[       OK ] PictureAddition/PictureAdditionTest.PictureAdditionTest/1 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAdditionTest/2
[       OK ] PictureAddition/PictureAdditionTest.PictureAdditionTest/2 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAdditionTest/3
[       OK ] PictureAddition/PictureAdditionTest.PictureAdditionTest/3 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAdditionTest/4
[       OK ] PictureAddition/PictureAdditionTest.PictureAdditionTest/4 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAdditionTest/5
[       OK ] PictureAddition/PictureAdditionTest.PictureAdditionTest/5 (1 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAdditionTest/6
[       OK ] PictureAddition/PictureAdditionTest.PictureAdditionTest/6 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAdditionTest/7
[       OK ] PictureAddition/PictureAdditionTest.PictureAdditionTest/7 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAdditionTest/8
[       OK ] PictureAddition/PictureAdditionTest.PictureAdditionTest/8 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAdditionTest/9
[       OK ] PictureAddition/PictureAdditionTest.PictureAdditionTest/9 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAdditionTest/10
[       OK ] PictureAddition/PictureAdditionTest.PictureAdditionTest/10 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAdditionTest/11
[       OK ] PictureAddition/PictureAdditionTest.PictureAdditionTest/11 (1 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAdditionTest/12
[       OK ] PictureAddition/PictureAdditionTest.PictureAdditionTest/12 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAdditionTest/13
[       OK ] PictureAddition/PictureAdditionTest.PictureAdditionTest/13 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAdditionTest/14
[       OK ] PictureAddition/PictureAdditionTest.PictureAdditionTest/14 (1 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/0
[       OK ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/0 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/1
[       OK ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/1 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/2
[       OK ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/2 (2 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/3
[       OK ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/3 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/4
[       OK ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/4 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/5
[       OK ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/5 (1 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/6
[       OK ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/6 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/7
[       OK ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/7 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/8
[       OK ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/8 (2 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/9
[       OK ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/9 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/10
[       OK ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/10 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/11
[       OK ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/11 (1 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/12
[       OK ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/12 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/13
[       OK ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/13 (0 ms)
[ RUN      ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/14
[       OK ] PictureAddition/PictureAdditionTest.PictureAddition16bitTest/14 (1 ms)
[----------] 30 tests from PictureAddition/PictureAdditionTest (44 ms total)

[----------] Global test environment tear-down
[==========] 30 tests from 1 test case ran. (53 ms total)
[  PASSED  ] 30 tests.


~~~

Cidana-Developers commented:
dry_run_test @./Bin/Release/SvtAv1UnitTests --gtest_filter=*PictureAddition*

ivy-zhang commented:
dry_run_test @./Bin/Release/SvtAv1UnitTests --gtest_filter=*PictureAddition*

Cidana-Developers commented on test/PictureAdditionTest.cc:
```diff
@@ -0,0 +1,301 @@
+                              uint32_t width, uint32_t height, int32_t bd);
+
+typedef struct PictureTestParam {
+    AreaSize areaSize;
```
rename areaSize to area_size

Cidana-Developers commented on test/PictureAdditionTest.cc:
```diff
@@ -0,0 +1,301 @@
+        test_size_ = MAX_PU_SIZE * MAX_PU_SIZE;
+    }
+
+	void SetUp() {
```
add override attributes

Cidana-Developers replied:
some format issue here.

Cidana-Developers commented on test/PictureAdditionTest.cc:
```diff
@@ -0,0 +1,301 @@
+        memset(recon2_, 0, test_size_ * sizeof(recon2_[0]));
+        memset(recon116bit_,
+               0,
+               test_size_ * sizeof(recon116bit_[0]));
```
can we rename recon116bit to recon1_16bit?

Cidana-Developers commented on test/PictureAdditionTest.cc:
```diff
@@ -0,0 +1,301 @@
+    }
+
+    void prepare_16bit_data() {
+        const int16_t mask = (1 << 8) - 1;
```
Is the mask correct? Should it changed to (1<<16) -1 ?

Cidana-Developers commented on test/PictureAdditionTest.cc:
```diff
@@ -0,0 +1,301 @@
+
+    void fail_pixel_count(uint32_t width, uint32_t height) {
+        int fail_Pixel_Count_ = 0;
+        for (uint16_t j = 0; j < height; j++) {
```
chang the uint16_t to uint32_t.

Cidana-Developers commented on test/PictureAdditionTest.cc:
```diff
@@ -0,0 +1,301 @@
+            break;
+        }
+        case REF_MAX: {
+            memset(residual_,
```
do not use memset here.

Cidana-Developers commented:
Can one of the admins verify this patch?

ivy-zhang commented:
dry_run_test @./Bin/Release/SvtAv1UnitTests --gtest_filter=*PictureAddition*

ivy-zhang commented:
dry_run_test @./Bin/Release/SvtAv1UnitTests --gtest_filter=*PictureAddition*

ivy-zhang commented:
dry_run_test @./Bin/Release/SvtAv1UnitTests --gtest_filter=*PictureAddition*

ivy-zhang commented:
dry_run_test @./Bin/Release/SvtAv1UnitTests --gtest_filter=*PictureAddition*

ivy-zhang commented:
dry_run_test @./Bin/Release/SvtAv1UnitTests --gtest_filter=*PictureAddition*

ivy-zhang commented:
dry_run_test @./Bin/Release/SvtAv1UnitTests --gtest_filter=*PictureAddition*

Cidana-Developers commented:
dry_run_test @./Bin/Release/SvtAv1UnitTests --gtest_filter=*PictureAddition*

ivy-zhang committed:
Compare with C implementation of Picture Addition

1. compare LbdPictureAdditionFunc and HbdPictureAdditionFunc with
   c implementation
2. fix a bug in run_16bit_test.

Cidana-Developers commented:
<!--
  0 Errors
  0 Warnings
  1 Message: Linux GCC Build PASS
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Linux GCC Build PASS</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

Cidana-Developers commented:
merged, close it.
